packages/vitest/src/node/core.ts

```diff
@@ -1097,6 +1097,9 @@ export class Vitest {
         if (reporter instanceof WebSocketReporter) {
           reporter.onFinishedReportCoverage()
         }
+        else if ('onFinishedReportCoverage' in reporter && typeof reporter.onFinishedReportCoverage === 'function') {
+          await reporter.onFinishedReportCoverage()
```

Comment:
should we find a better way to do this?

Reply:
Is it possible to check `reporter instanceof HTMLReporter` just like `WebSocketReporter` above?

In https://github.com/vitest-dev/vitest/pull/5242#discussion_r1495440749, we discussed and we didn't go with `onFinishedReportCoverage` at `interface Reporter` level for now (though it might change in reporter API rework https://github.com/vitest-dev/vitest/issues/6147).

Reply:
`HTMLReporter` is defined in `@vitest/ui`, not in `vitest`, so I don't think we can do that

Reply:
How about adding a new type like BuildReporter?